src/background/legacy-external-message-handler.ts

```diff
@@ -148,10 +158,13 @@ export async function handleLegacyExternalMethodFormat(
     }
 
     case ExternalMethods.structuredDataSignatureRequest: {
-      const { urlParams, tabId } = makeSearchParamsWithDefaults(port, [
+      const otherParams: OtherParams = [
         ['request', payload],
         ['messageType', 'structured'],
-      ]);
+        ['flow', ExternalMethods.structuredDataSignatureRequest],
+      ];
+      if (port.sender?.origin) otherParams.push(['referringAppDomain', port.sender.origin]);
```

Comment:
We already pass through the `origin` param. This is already handled in `makeSearchParamsWithDefaults`, so can remove all `referringAppDomain` references.